File: 1266.minimum-time-visiting-all-points/solution.kt
class Solution {
    fun minTimeToVisitAllPoints(points: Array<IntArray>): Int {
        return (1 until points.size).sumOf { i ->
            val dx = abs(points[i - 1][0] - points[i][0])
            val dy = abs(points[i - 1][1] - points[i][1])
            maxOf(dx, dy)
        }
    }
}

File: 1266.minimum-time-visiting-all-points/solution.go
func minTimeToVisitAllPoints(points [][]int) int {
    res := 0
    for i := 1; i < len(points); i++ {
        dx := abs(points[i][0] - points[i-1][0])
        dy := abs(points[i][1] - points[i-1][1])
        if dx > dy {
            res += dx
        } else {
            res += dy
        }
    }
    return res
}

func abs(a int) int {
    if a < 0 {
        return -a
    }
    return a
}